File: En Go - Compilado/SumaSubconjuntos.go
package main 
 
import ( 
    "fmt" 
)

func capturar(tam uint32) []uint32 { 
    conjunto := make([]uint32, 0) 
    var sb, num uint32 
    for sb = 0; sb < tam; sb++ { 
        fmt.Printf("Ingrese el numor #%d: ", sb + 1) 
        fmt.Scanf("%d", &num) 
        conjunto = append(conjunto, num) 
    } 
    return conjunto 
} 

func sumar(conjunto []uint32, tam uint32) (uint32, uint32) { 
    var sumsub1, sumsub2, reban uint32 = 0, 0, 0 
    var sb1, sb2 uint32 

    reban = uint32(len(conjunto[0:uint32(len(conjunto) / 2)])) 
    for sb1 = 0; sb1 < reban; sb1++ { 
        sumsub1 = sumsub1 + conjunto[sb1] 
    } 
    for sb2 = reban; sb2 < tam; sb2++ { 
        sumsub2 = sumsub2 + conjunto[sb2] 
    } 
    return sumsub1, sumsub2 
} 

func main() { 
    conjunto := make([]uint32, 0) 
    var tam, sumsub1, sumsub2 uint32 

    fmt.Print("Ingrese el numero de valores del conjunto: ") 
    fmt.Scanf("%d", &tam) 

    fmt.Println() 
    conjunto = capturar(tam) 
    sumsub1, sumsub2 = sumar(conjunto, tam) 

    if sumsub1 == sumsub2 { 
        fmt.Println("La suma de los subconjuntos da el mismo resultado!") 
    } else { 
        fmt.Println("La suma de los subconjuntos no da el mismo resultado!") 
    } 
} 

File: En C++ - Compilado/SumaSubconjuntos.cpp
#include <iostream>
#include <new>
using namespace std;

void capturar(unsigned int *conjunto[], unsigned int &tam) {
    unsigned int num, sb;
    for(sb = 0; sb < tam; sb++) {
        cout << "Ingrese el valor #" << sb + 1 << ": ";
        cin >> num;
        (*conjunto)[sb] = num;
    }
}

int main() {
    unsigned int *conjunto;
    unsigned int tam, reban, sbn;
    unsigned int sbnsum1 = 0, sbnsum2 = 0;

    cout << "Ingrese el numero de valores del conjunto: ";
    cin >> tam;

    cout << "\n";
    conjunto = new unsigned int[tam];
    capturar(&conjunto, tam);

    reban = tam / 2;

    for(sbn = 0; sbn < reban; sbn++) {
        sbnsum1 += conjunto[sbn];
    }

    for(sbn = reban; sbn < tam; sbn++) {
        sbnsum2 += conjunto[sbn];
    }

    cout << "\n";
    if(sbnsum1 == sbnsum2) {
        cout << "La suma de los subconjuntos da el mismo resultado!" << endl;
    } else {
        cout << "La suma de los subconjuntos no da el mismo resultado!" << endl;
    }

    delete[] conjunto;
    return 0;
}
